cpp/tests/join/cross_join_tests.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/join.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_view.hpp>
+
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/table_utilities.hpp>
+
+template <typename T>
+using column_wrapper = cudf::test::fixed_width_column_wrapper<T>;
+
+struct JoinTest : public cudf::test::BaseFixture {
+};
+
+TEST_F(JoinTest, CrossJoin)
+{
+  auto a_strings  = std::vector<const char*>{"quick", "accénted", "turtlé", "composéd"};
+  auto b_strings  = std::vector<const char*>{"result", "", "words"};
+  auto e_strings3 = std::vector<const char*>{"quick",
+                                             "quick",
+                                             "quick",
+                                             "accénted",
+                                             "accénted",
+                                             "accénted",
+                                             "turtlé",
+                                             "turtlé",
+                                             "turtlé",
+                                             "composéd",
+                                             "composéd",
+                                             "composéd"};
+  auto e_strings7 = std::vector<const char*>{
+    "result", "", "words", "result", "", "words", "result", "", "words", "result", "", "words"};
+
+  auto a_0 = column_wrapper<int32_t>{10, 20, 20, 50};
+  auto a_1 = column_wrapper<float>{5.0, .5, .5, .7};
+  auto a_2 = column_wrapper<int8_t>{90, 77, 78, 41};
+  auto a_3 = cudf::test::strings_column_wrapper(
```

Comment:
There's no need to define the `vector<const char*>` and then use it to initialize the wrapper. You can just initialize the wrapper directly:
```
strings_column_wrapper s( {"quick", "accénted", "turtlé", "composéd"} );
```